libraries/AP_HAL_ChibiOS/hwdef/common/stm32f47_mcuconf.h

```diff
@@ -580,3 +580,10 @@
 
 // limit ISR count per byte
 #define STM32_I2C_ISR_LIMIT                 6
+
+#if defined(STM32F7xx_MCUCONF)
```

Comment:
I'm not convinced this is correct and worryingly we do it differently in this mcuconf to all the others. I think this symbol is supposed to be defined in the mcuconf defining the mcuconf that is supported. We do that in all the others, but not this one. More worryingly because this is shared we should be defining both this and STM32F4xx_MCUCONF which would make this check invalid. I think instead you should be checking for STM32F7xx

Reply:
I chose it as it is what is used earlier in the same header file for selecting between F4 and F7 clock trees. So we can be sure it is a correct check as if it wasn't then the boards wouldn't compile/boot (bad clock tree)

Reply:
Ok, so perhaps a cleanup for a different PR. This wouldn't work in other mcuconf which is highly confusing.